File: src/PAT/P1012.java
package PAT;

import java.util.*;
import java.io.*;

class Stu {
	String ID;
	int C;
	int M;
	int E;
	int A;
	int Arank;
	public Stu(String ID, int C, int M, int E) {
		this.ID = ID;
		this.C = C;
		this.M = M;
		this.E = E;
		this.A = (C+M+E)/3;
	}
}

public class P1012 {
	static void getRank(ArrayList<Stu> list, String ID) {
		int Crank = 1;
		int Mrank = 1;
		int Erank = 1;
		int Arank = 1;
		Stu currentStu = null;
		for(int i=0; i<list.size(); i++) {
			if(ID.equals(list.get(i).ID)) {
				currentStu = list.get(i);
				break;
			}
		}
		if(currentStu == null) {
			System.out.println("N/A");
			return ;
		}
			
		for(int i=0; i<list.size(); i++) {
			if(list.get(i).C > currentStu.C) 
				Crank ++;
		}
		for(int i=0; i<list.size(); i++) {
			if(list.get(i).M > currentStu.M) 
				Mrank ++;
		}
		for(int i=0; i<list.size(); i++) {
			if(list.get(i).E > currentStu.E) 
				Erank ++;
		}
		for(int i=0; i<list.size(); i++) {
			if(list.get(i).A > currentStu.A) 
				Arank ++;
		}
		
		int best = Math.min(Crank, Math.min(Mrank, Math.min(Erank, Arank)));
		if(Arank == best) {
			System.out.println(best + " " + "A");
			return ;
		}
		if(Crank == best) {
			System.out.println(best + " " + "C");
			return ;
		}
		if(Mrank == best) {
			System.out.println(best + " " + "M");
			return ;
		}
		if(Erank == best) {
			System.out.println(best + " " + "E");
			return ;
		}
	}
	
	public static void main(String[] args) {
		Scanner sc = new Scanner(new BufferedInputStream(System.in));
		ArrayList<Stu> list = new ArrayList<>();
		int N = sc.nextInt();
		int M = sc.nextInt();
		for(int i=0; i<N; i++) {
			list.add(new Stu(sc.next(), sc.nextInt(), sc.nextInt(), sc.nextInt()));
		}
		
		for(int i=0; i<M; i++) {
			getRank(list, sc.next());
		}
		
		sc.close();

	}

}

File: src/PAT/P1012.cpp
#include<bits/stdc++.h> 

using namespace std;

struct Stu {
	int ID;
	int C;
	int M;
	int E;
	int A;
};

void getRank(vector<Stu> vec, int ID);

vector<Stu> vec; 

int main() {
	int N, M;
	int ID;
	int CG, MG, EG;
	cin >> N >> M;
	Stu stu;
	for(int i=0; i<N; i++) {
		cin >> ID;
		stu.ID = ID;
		cin >> CG;
		stu.C = CG;
		cin >> MG;
		stu.M = MG;
		cin >> EG;
		stu.E = EG;
		stu.A = (CG+MG+EG)/3;
		vec.push_back(stu);
	}
	
	for(int i=0; i<M; i++) {
		cin >> ID;
		getRank(vec, ID);
	}
		
	return 0;
}

void getRank(vector<Stu> vec, int ID) {
	int Crank = 1;
	int Mrank = 1;
	int Erank = 1;
	int Arank = 1;
	Stu currentStu;
	currentStu.ID = -1;
	for(int i=0; i<vec.size(); i++) {
		if(ID == vec[i].ID) {
			currentStu = vec[i];
			break;
		}
	}
	if(currentStu.ID == -1) {
		cout << "N/A" << endl;
		return ;
	}
		
	for(int i=0; i<vec.size(); i++) {
		if(vec[i].C > currentStu.C) 
			Crank ++;
	}
	for(int i=0; i<vec.size(); i++) {
		if(vec[i].M > currentStu.M) 
			Mrank ++;
	}
	for(int i=0; i<vec.size(); i++) {
		if(vec[i].E > currentStu.E) 
			Erank ++;
	}
	for(int i=0; i<vec.size(); i++) {
		if(vec[i].A > currentStu.A) 
			Arank ++;
	}
	
	int best = min(Crank, min(Mrank, min(Erank, Arank)));
	if(Arank == best) {
		cout << best << " " << "A" << endl;
		return ;
	}
	if(Crank == best) {
		cout << best << " " << "C" << endl;
		return ;
	}
	if(Mrank == best) {
		cout << best << " " << "M" << endl;
		return ;
	}
	if(Erank == best) {
		cout << best << " " << "E" << endl;
		return ;
	}
}
